src/commons/utils/corio_utils.py

```diff
@@ -265,6 +275,39 @@ def setup_environment():
     os.makedirs(DATA_DIR_PATH, exist_ok=True)
 
 
+async def activate_degraded_mode() -> None:
+    """Activate degraded mode for execution by shutdown/deleting pod."""
+    bucket = f'object-op-{time.perf_counter_ns()}'.lower()
+    access_key = literal_eval(os.getenv("options")).get("access_key")
+    secret_key = literal_eval(os.getenv("options")).get("secret_key")
+    endpoint_url = literal_eval(os.getenv("options")).get("endpoint_url")
+    use_ssl = literal_eval(os.environ.get("options")).get("use_ssl")
+    response = await S3Bucket(access_key, secret_key, endpoint_url=endpoint_url,
+                              use_ssl=use_ssl).create_bucket(bucket)
+    if response:
+        os.environ["d_bucket"] = bucket
+    nodes = CLUSTER_CFG["nodes"]
+    (host, user, password) = (None, None, None)
+    for node in nodes:
+        if node["node_type"] == "master":
+            host = node["hostname"]
+            user = node["username"]
+            password = node["password"]
+            break
+    logical_node = LogicalNode(host=host,
+                               user=user,
+                               password=password)
+    LOGGER.info("Shutdown the data pod safely by making replicas=0")
+    LOGGER.info("Get pod name to be deleted")
+    pod_list = logical_node.get_all_pods(pod_prefix=DATA_POD_NAME_PREFIX)
+    pod_name = random.sample(pod_list, 1)[0]
+    hostname = logical_node.get_pod_hostname(pod_name=pod_name)
+    LOGGER.info("Deleting pod %s", pod_name)
+    resp = logical_node.create_pod_replicas(num_replica=0, pod_name=pod_name)
```

Comment:
Are we taking care of IO failures happening during pod shutdown .

Reply:
No, As of now this implementation is for testing degraded mode which is known in advance, hence if user select test mode as degraded mode and gives how many nodes need to be degreded then , it creates bucket in advance and shuts the pods down and adjust the no. of sessions and starts IO on degraded setup with already created bucket.

Reply:
@VikasKumarC  As of now inflight IO's are not supported for F-99E. 
https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/968622186/F-99E+IO+stability+in+degraded+mode+up+to+7+days+single+node+failure

So implementation needs to take care of not performing any IO operations during pod shutdown. 

cc @sonalk0209 @pranav-sahasrabudhe @seagate-sarang-sawant